newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudResourceMover.js

```diff
@@ -161,15 +162,113 @@ export const moveAllCloudProjectResourcesToCloudProject = async ({
   uploadedProjectResourceFiles.forEach(({ url, error }, index) => {
     const resource = downloadedFilesAndResourcesToUpload[index].resource;
     if (error || !url) {
-      result.erroredResources.push({
-        resourceName: resource.getName(),
-        error: error || new Error('Unknown error during upload.'),
-      });
+      onError(
+        resource.getName(),
+        error || new Error('Unknown error during upload.')
+      );
       return;
     }
 
     resource.setFile(url);
   });
+};
+
+export const moveAllCloudProjectResourcesToCloudProject = async ({
+  project,
+  authenticatedUser,
+  oldFileMetadata,
+  newFileMetadata,
+  oldStorageProvider,
+  oldStorageProviderOperations,
+  newStorageProvider,
+  newStorageProviderOperations,
+  onProgress,
+}: MoveAllProjectResourcesOptions): Promise<MoveAllProjectResourcesResult> => {
+  const result: MoveAllProjectResourcesResult = {
+    erroredResources: [],
+  };
+
+  const newCloudProjectId = newFileMetadata.fileIdentifier;
+
+  const resourcesToFetchAndUpload = getResourcesToFetchAndUpload({
+    project,
+    selectCondition: checkIfIsGDevelopCloudBucketUrl,
+    ignoreCondition: fileName =>
+      extractProjectUuidFromProjetResourceUrl(fileName) === newCloudProjectId,
+    onError: (resourceName, errorMessage) => {
+      result.erroredResources.push({
+        resourceName,
+        error: new Error(errorMessage),
+      });
+    },
+  });
+
+  // If an error happens here, it will be thrown out of the function.
+  if (oldStorageProviderOperations.onEnsureCanAccessResources)
+    oldStorageProviderOperations.onEnsureCanAccessResources(
+      project,
+      oldFileMetadata
+    );
+
+  await uploadResourcesToCloud({
+    resourcesToFetchAndUpload,
+    authenticatedUser,
+    cloudProjectId: newCloudProjectId,
+    onProgress,
+    onError: (resourceName, error) => {
+      result.erroredResources.push({
+        resourceName,
+        error,
+      });
+    },
+  });
+
+  return result;
+};
+
+export const moveUrlResourcesToCloudFilesIfPrivate = async ({
```

Comment:
See my other comment - what you're doing here is not a "move" of resources between 2 different projects, but a "fetch" of resources inside a single project.
So this should live inside a XxxResourceFetcher file.

Reply:
It did feel like a move, but of resources, rather than of a project.
The code is gonna be incredibly similar